synapse/storage/databases/main/registration.py

```diff
@@ -166,6 +166,42 @@ async def get_user_by_id(self, user_id: str) -> Optional[Dict[str, Any]]:
             desc="get_user_by_id",
         )
 
+    @cached()
```

Comment:
it'd be swell if `get_userinfo_by_id` and `get_user_by_id` could share a cache. I think the easiest way to do that is by having one call the other.